python/ray/data/_internal/arrow_block.py

```diff
@@ -175,6 +175,13 @@ def _build_tensor_row(row: ArrowRow) -> np.ndarray:
         # Getting an item in a tensor column automatically does a NumPy conversion.
         return row[VALUE_COL_NAME][0]
 
+    def get_keys(self, key: KeyFn) -> Optional[np.ndarray]:
+        # The key must be a str for Arrow table.
+        assert isinstance(key, str)
```

Comment:
key can be None?

Reply:
```
 # Always allow None since groupby interprets that as grouping all
        # records into a single global group.
        if key is not None:
            _validate_key_fn(self, key)
```

None is possible for arrow otherwise we cannot support global aggregation (i.e. ds.aggregate())

Reply:
I see, we should centralize the key validation into one place (e.g. _validate_key_fn) to make it more clear. Will file a bug for that cleanup.